src/vos/evtree.c

```diff
@@ -2746,12 +2730,33 @@ evt_common_rect_weight(struct evt_context *tcx, const struct evt_rect *rect,
 	return 0;
 }
 
+void
+evt_split_common(struct evt_context *tcx, bool leaf, struct evt_node *nd_src,
+		 struct evt_node *nd_dst, int idx)
+{
+	void	*entry_src;
+	void	*entry_dst;
+	size_t	 entry_size;
+
+	if (leaf) {
+		entry_src = evt_node_entry_at(tcx, nd_src, idx);
+		entry_dst = evt_node_entry_at(tcx, nd_dst, 0);
+		entry_size = sizeof(struct evt_node_entry);
+	} else {
+		entry_src = &nd_src->tn_child[idx];
+		entry_dst = &nd_dst->tn_child[0];
+		entry_size = sizeof(nd_dst->tn_child[0]);
+	}
+
+	memcpy(entry_dst, entry_src, entry_size * (nd_src->tn_nr - idx));
+	nd_dst->tn_nr = nd_src->tn_nr - idx;
```

Comment:
(style)  'nd' may be misspelled - perhaps 'and'?
(style)  'nd' may be misspelled - perhaps 'and'?